service/src/main/java/bio/terra/workspace/service/workspace/flight/DeleteControlledSamResourcesStep.java

```diff
@@ -0,0 +1,82 @@
+package bio.terra.workspace.service.workspace.flight;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.iam.SamService;
+import bio.terra.workspace.service.resource.controlled.ControlledResource;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys;
+import java.util.List;
+
+/**
+ * A step which deletes the Sam data of the controlled resources provided by {@link
+ * ListControlledResourceIdsStep} Multiple deletions are run in parallel as resources are
+ * independent.
+ */
+public class DeleteControlledSamResourcesStep implements Step {
+
+  private final SamService samService;
+  private final AuthenticatedUserRequest userRequest;
+
+  public DeleteControlledSamResourcesStep(
+      SamService samService, AuthenticatedUserRequest userRequest) {
+    this.samService = samService;
+    this.userRequest = userRequest;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext)
+      throws InterruptedException, RetryException {
+    FlightMap workingMap = flightContext.getWorkingMap();
+
+    @SuppressWarnings("unchecked")
+    List<ControlledResource> controlledResourceList =
+        workingMap.get(ControlledResourceKeys.CONTROLLED_RESOURCE_LIST, List.class);
+
+    try {
+      // Deleting each resource takes a call to Sam. As there may be many many resources and they
+      // are all independent, running these calls in parallel can save time.
+      controlledResourceList.parallelStream()
+          .forEach(
+              controlledResource -> {
+                try {
+                  samService.deleteControlledResource(controlledResource, userRequest);
+                } catch (InterruptedException e) {
+                  // This function must handle the checked InterruptedException, even though the
+                  // Step as a whole is fine surfacing it. Here, we wrap the InterruptedException in
+                  // an unchecked exception, smuggle it out, then catch and throw it later.
+                  throw new UncheckedInterruptedException(e);
+                }
+              });
+    } catch (UncheckedInterruptedException e) {
+      throw e.getException();
```

Comment:
Can Sam implement a batch delete endpoint? This is a lot of round trips.

Reply:
it's definitely a possible future optimization. If performance becomes an issue here, it will be worth a discussion with CI on how we should send many requests.